File: backend/app/models/enrollment.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime
from ..database import Base

# In your Enrollment model
class Enrollment(Base):
    __tablename__ = "enrollments"

    id = Column(Integer, primary_key=True, index=True)
    employee_id = Column(Integer, ForeignKey("employees.id"))
    training_id = Column(Integer, ForeignKey("trainings.id"))
    status = Column(String(20), default="enrolled")  # enrolled, in_progress, completed, cancelled
    progress = Column(Integer, default=0)  # Percentage 0-100
    enrolled_date = Column(DateTime, default=datetime.utcnow)
    start_date = Column(DateTime)
    end_date = Column(DateTime)
    completed_date = Column(DateTime)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    employee = relationship("Employee", back_populates="enrollments")
    training = relationship("Training", back_populates="enrollments")

File: frontend/src/types/enrollment.ts
// enrollment.ts
export interface Enrollment {
    id: number;
    employee_id: number;
    training_id: number;
    status: "enrolled" | "in_progress" | "completed" | "cancelled";
    progress: number;
    enrolled_date: string; // This was missing
    start_date: string | null;
    end_date: string | null;
    completed_date: string | null;
    created_at: string;
    updated_at: string;
    // Optional relationships (if your API includes them)
    employee?: {
        id: number;
        first_name: string;
        last_name: string;
        email: string;
        department?: {
            id: number;
            name: string;
        };
    };
    training?: {
        id: number;
        name: string;
        description: string;
        duration_hours: number;
    };
}

export interface EnrollmentFormData {
    employee_id: number | "";
    training_id: number | "";
    status: "enrolled" | "in_progress" | "completed" | "cancelled";
    progress?: number;
    start_date: string;
    end_date: string;
    // Note: enrolled_date, created_at, updated_at are set by the server
}

// For update operations
export interface EnrollmentUpdateData {
    employee_id?: number;
    training_id?: number;
    status?: "enrolled" | "in_progress" | "completed" | "cancelled";
    progress?: number;
    start_date?: string;
    end_date?: string;
    completed_date?: string | null;
}

export interface TrainingProgressItem {
    id: string;
    name: string;
    role: string;
    avatarUrl?: string;
    trainingName: string;
    progress?: number;
    status: "enrolled" | "in_progress" | "completed" | "cancelled" | "overdue";
    startDate?: string;
    endDate?: string;
    deadline?: string;
    completionDate?: string;
    hasCertification?: boolean;
}
